src/prefect/utilities/notifications.py

```diff
@@ -279,5 +292,83 @@ def add(x, y):
     form_data = slack_message_formatter(tracked_obj, new_state)
     r = requests.post(webhook_url, json=form_data)
     if not r.ok:
-        raise ValueError("Slack notification for {} failed".format(tracked_obj))
+        raise ValueError(
+            "Slack notification for {} failed".format(tracked_obj))
+    return new_state
+
+
+@curry
+def jira_notifier(
+    tracked_obj: TrackedObjectType,
+    old_state: "prefect.engine.state.State",
+    new_state: "prefect.engine.state.State",
+    ignore_states: list = None,
+    only_states: list = None,
+    projectName: str = None,
+    assignee: str = -1
+) -> "prefect.engine.state.State":
+    """
+    Slack state change handler; requires having the Prefect slack app installed.
+    Works as a standalone state handler, or can be called from within a custom
+    state handler.  This function is curried meaning that it can be called multiple times to partially bind any keyword arguments (see example below).
+
+    Args:
+        - tracked_obj (Task or Flow): Task or Flow object the handler is
+            registered with
+        - old_state (State): previous state of tracked object
+        - new_state (State): new state of tracked object
+        - ignore_states ([State], optional): list of `State` classes to ignore,
+            e.g., `[Running, Scheduled]`. If `new_state` is an instance of one of the passed states, no notification will occur.
+        - only_states ([State], optional): similar to `ignore_states`, but
+            instead _only_ notifies you if the Task / Flow is in a state from the provided list of `State` classes
+        - webhook_secret (str, optional): the name of the Prefect Secret that stores your slack webhook URL;
+            defaults to `"SLACK_WEBHOOK_URL"`
+
+    Returns:
+        - State: the `new_state` object that was provided
+
+    Raises:
+        - ValueError: if the slack notification fails for any reason
+
+    Example:
+        ```python
+        from prefect import task
+        from prefect.utilities.notifications import slack_notifier
+
+        @task(state_handlers=[slack_notifier(ignore_states=[Running])]) # uses currying
+        def add(x, y):
+            return x + y
+        ```
+    """
+    username = cast(str, prefect.client.Secret("JIRAUSER").get())
+    password = cast(str, prefect.client.Secret("JIRATOKEN").get())
+    serverURL = cast(str, prefect.client.Secret("JIRASERVER").get())
+
+    if not projectName:
+        projectName = cast(str, prefect.client.Secret("JIRAPROJECT").get())
```

Comment:
Two thoughts on the use of secrets here:
- outside of the scope of this PR, it would be nice to find a way around _requiring_ Prefect Cloud Secrets
- since you require 4 distinct secrets, you might consider refactoring this to one `JIRA_CREDS` secret which is a JSON payload (a dictionary in Python) with these four keys; might make the setting of the secret easier on the user (that is also just my own personal preference - open to push back)

Reply:
I like Chris's suggestion here